buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -71,6 +76,11 @@
 
     private final LinkedHashMap<String, Predicate<ElasticsearchNode>> waitConditions;
     private final List<URI> plugins = new ArrayList<>();
+    // Must use Object here to retain compatibility with the Groovy DSL in Gradle (GString)
```

Comment:
If we want to retain compatibility with `GString` yet keep some type safety then we can use `CharSequence` instead.